src/public/cards/edgeMapping.ts

```diff
@@ -0,0 +1,140 @@
+import { CategoriesMapping } from "../models";
+import * as utils from "../utils";
+
+export class CSVEdgeMapping {
+  private container!: HTMLElement;
+  private inputs!: HTMLCollectionOf<HTMLInputElement>;
+  private importButton!: HTMLButtonElement;
+
+  init() {
+    this.container = document.getElementsByClassName(
+      "edgeMappingContainer"
+    )[0] as HTMLElement;
+    this.inputs = this.container.getElementsByClassName(
+      "mapInput"
+    ) as HTMLCollectionOf<HTMLInputElement>;
+    this.importButton = this.container.getElementsByClassName(
+      "primaryButton"
+    )[0] as HTMLButtonElement;
+    this.importButton.disabled = true;
+    this.hideCard();
+
+    this.inputs[0].value = "";
+    this.inputs[1].value = "";
+    this.inputs[0].addEventListener("input", this.onChangeInput.bind(this));
+    this.inputs[1].addEventListener("input", this.onChangeInput.bind(this));
+  }
+
+  onChangeInput() {
+    this.importButton.disabled = !(
+      this.inputs[0].value && this.inputs[1].value
+    );
+  }
+
+  /**
+   * Using data in session storage, import it and return message of success
+   */
+  async importEdges(categoriesMapping: CategoriesMapping): Promise<string> {
+    utils.startWaiting();
+    try {
+      const rows = sessionStorage.getItem("rows");
+      const headers = sessionStorage.getItem("headers");
+      const edgeType = sessionStorage.getItem("entityName");
+      if (rows && headers && edgeType) {
+        const rowsParsed = JSON.parse(rows);
+        const headersParsed = headers.split(",");
+        const queryTemplate = this.createEdgeTemplate(
+          categoriesMapping,
+          edgeType,
+          headersParsed
+        );
+        const edges = this.createQueries(queryTemplate, rowsParsed);
+        const resNodes = await utils.makeRequest(
+          "POST",
+          `api/addEdges?sourceKey=${sessionStorage.getItem("sourceKey")}`,
+          {
+            edges: edges,
+          }
+        );
+        const data = JSON.parse(resNodes.response);
+        return `${data.success}/${data.total} edges have been added to the database`;
+      }
+      return "";
+    } catch (error) {
+      throw new Error("Import has failed");
+    } finally {
+      utils.stopWaiting();
+    }
+  }
+
+  /**
+   * From the edge config create query templates
+   */
+  private createEdgeTemplate(
```

Comment:
not possible because we are not using the node ids (but properties). Possibly we will query the node ids first and then use the restclient, part of next iteration